### PR TITLE
Wait conditions

### PR DESCRIPTION
**Pull request by willthames**
##### SUMMARY
Improve k8s Deployment and Daemonset wait conditions
    
Ensure that Deployments and Daemonsets properly await
all replicas to be available. Correctly handles the
subtle edge case when a service account no longer exists.
    
Note that this will dramatically slow Daemonset updates

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
k8s

### CODE REVIEW

**Comment by willthames** [comment]
Tests failing would be fixed by #34

File: molecule/default/tasks/apply.yml [hunk 1]
```diff
@@ -179,6 +182,227 @@
...
+      k8s_scale:
+        api_version: apps/v1
+        kind: Deployment
+        name: scale-deploy
+        namespace: "{{ apply_namespace }}"
+        replicas: 0
+
+    - name: wait for pods to scale down
+      pause:
+        seconds: 30
```
Comment (fabianvf) [inline molecule/default/tasks/apply.yml]: can we use the `wait` field for this scale down?

File: plugins/module_utils/scale.py [hunk 2]
```diff
@@ -29,11 +29,16 @@
...
-    from openshift import watch
     from openshift.dynamic.client import ResourceInstance
-    from openshift.helper.exceptions import KubernetesException
-except ImportError as exc:
-    class KubernetesException(Exception):
+    from openshift.dynamic.exceptions import NotFoundError
+except ImportError:
+    pass
+try:
+    from openshift import watch
```
Comment (fabianvf) [inline plugins/module_utils/scale.py]: I don't think we have a `watch` module in the openshift client anymore, we just make use of the `kubernetes` `watch` module instead. Only way to call a watch from the openshift client is the `DynamicClient.watch` method, though using `watch` from `kubernetes` should work fine.

File: plugins/module_utils/scale.py [hunk 3]
```diff
@@ -189,15 +194,12 @@ def _create_stream(self, resource, namespace, wait_time):
...
-            w._api_client = self.client.client
-            if namespace:
-                stream = w.stream(resource.get, serialize=False, namespace=namespace, timeout_seconds=wait_time)
-            else:
-                stream = w.stream(resource.get, serialize=False, namespace=namespace, timeout_seconds=wait_time)
-        except KubernetesException:
-            pass
+        w = watch.Watch()
+        w._api_client = self.client.client
+        if namespace:
```
Comment (fabianvf) [inline plugins/module_utils/scale.py]: It looks like these two branches are identical, we can probably just remove this check.

File: molecule/default/tasks/apply.yml (hunk 1 above)
Comment (willthames) [reply molecule/default/tasks/apply.yml]: I'm moving this to a separate PR for scale testing

File: plugins/module_utils/scale.py (hunk 2 above)
Comment (willthames) [reply plugins/module_utils/scale.py:37]: I'm moving this to a separate PR for scale improvements

File: plugins/module_utils/scale.py (hunk 3 above)
Comment (willthames) [reply plugins/module_utils/scale.py:199]: I'm moving this to a separate PR for scale testing

**Pull request merged**